tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,375 @@
+            assert config_dict[key] == value
+
+        # NOTE: This 'name' is the qualname of the class, which is used by the Factory metaclass'
+        # `__call__` method.
+        name = self.Task.__qualname__
+        from orion.benchmark.benchmark_client import _get_task
+        second_task = _get_task(name=name, **config_dict)
+        assert second_task.configuration == first_task.configuration        
+        
+
+    @pytest.mark.xfail(
+        reason="TODO: Not sure how to modify the `space.sample` for a particular space instance "
+        "without changing the 'global' RNG."
+    )
+    def test_space_sample_is_reproducible(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        task = self.Task(
+            max_trials=10,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        points = task._space.sample(10)
+        assert len(set(points)) == 10
+
+        # Create a new task:
+        second_task = self.Task(
+            max_trials=10,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+        assert second_task._space.sample(10) == points
+    
+    def test_call_is_reproducible(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        """ Two tasks created with the same args, given the same point, should produce the same
+        results.
+        """
+        task_id = 0
+        first_task = self.Task(
+            max_trials=10,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+
+        point = first_task._space.sample(1, seed=first_task.seed)[0]
+
+        second_task = self.Task(
+            max_trials=10,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+
+        results = second_task(point)
+
+        assert len(results) == 1
+        assert results[0]["type"] == "objective"
+        first_objective = results[0]["value"]
+        second_results = second_task(point)
+        second_objective = second_results[0]["value"]
+
+        assert second_objective == first_objective
+
+    def test_call_twice_with_same_task_gives_same_result(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        """ When using the same task and different values for the same point, the results should
+        be identical.
```

Comment:
You are testing both reproducibility of tasks call and the support for different types of `x`. I don't think you need to test reproducibility here, just that the objectives are the same across different types.

Reply:
Sorry to ask, but since I am checking that the objectives are the same, can I just leave this as-is?